

class Ui_CreateProduct(object):
    def __init__(self):
        self.create_product_controller = CreateProductController(self)
        self.product = Product(connection())
   
   
    def setupUi(self, CreateProduct):
        CreateProduct.setObjectName("CreateProduct")
        CreateProduct.resize(396, 298)
        self.label_6 = QtWidgets.QLabel(CreateProduct)
        self.label_6.setGeometry(QtCore.QRect(0, 0, 400, 300))
        self.label_6.setText("")
        self.label_6.setPixmap(QtGui.QPixmap("Imagenes/tweety2.jpg"))
        self.label_6.setObjectName("label_6")
        self.input_cod = QtWidgets.QLineEdit(CreateProduct)
        self.input_cod.setGeometry(QtCore.QRect(140, 50, 167, 25))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_cod.setFont(font)
        self.input_cod.setText("")
        self.input_cod.setObjectName("input_cod")
        self.label_cod = QtWidgets.QLabel(CreateProduct)
        self.label_cod.setGeometry(QtCore.QRect(30, 60, 73, 19))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_cod.setFont(font)
        self.label_cod.setObjectName("label_cod")
        self.input_cant = QtWidgets.QLineEdit(CreateProduct)
        self.input_cant.setGeometry(QtCore.QRect(140, 79, 167, 25))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_cant.setFont(font)
        self.input_cant.setText("")
        self.input_cant.setObjectName("input_cant")
        self.btn_create = QtWidgets.QPushButton(CreateProduct)
        self.btn_create.setGeometry(QtCore.QRect(40, 210, 301, 27))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.btn_create.setFont(font)
        self.btn_create.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
        self.btn_create.setObjectName("btn_create")
        self.label_descrip = QtWidgets.QLabel(CreateProduct)
        self.label_descrip.setGeometry(QtCore.QRect(30, 120, 101, 19))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_descrip.setFont(font)
        self.label_descrip.setObjectName("label_descrip")
        self.label = QtWidgets.QLabel(CreateProduct)
        self.label.setGeometry(QtCore.QRect(41, 1, 300, 33))
        font = QtGui.QFont()
        font.setPointSize(20)
        font.setBold(True)
        font.setWeight(75)
        self.label.setFont(font)
        self.label.setObjectName("label")
        self.label_categoria = QtWidgets.QLabel(CreateProduct)
        self.label_categoria.setGeometry(QtCore.QRect(30, 182, 86, 19))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_categoria.setFont(font)
        self.label_categoria.setObjectName("label_categoria")
        self.btn_exit = QtWidgets.QPushButton(CreateProduct)
        self.btn_exit.setGeometry(QtCore.QRect(40, 270, 301, 27))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.btn_exit.setFont(font)
        self.btn_exit.setObjectName("btn_exit")
        self.label_precio = QtWidgets.QLabel(CreateProduct)
        self.label_precio.setGeometry(QtCore.QRect(30, 151, 57, 19))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_precio.setFont(font)
        self.label_precio.setObjectName("label_precio")
        self.input_categoria = QtWidgets.QLineEdit(CreateProduct)
        self.input_categoria.setGeometry(QtCore.QRect(140, 172, 167, 25))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_categoria.setFont(font)
        self.input_categoria.setObjectName("input_categoria")
        self.btn_cancel = QtWidgets.QPushButton(CreateProduct)
        self.btn_cancel.setGeometry(QtCore.QRect(40, 240, 301, 27))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.btn_cancel.setFont(font)
        self.btn_cancel.setObjectName("btn_cancel")
        self.input_descrip = QtWidgets.QLineEdit(CreateProduct)
        self.input_descrip.setGeometry(QtCore.QRect(140, 110, 167, 25))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_descrip.setFont(font)
        self.input_descrip.setObjectName("input_descrip")
        self.input_precio = QtWidgets.QLineEdit(CreateProduct)
        self.input_precio.setGeometry(QtCore.QRect(140, 141, 167, 25))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.input_precio.setFont(font)
        self.input_precio.setObjectName("input_precio")
        self.label_cant = QtWidgets.QLabel(CreateProduct)
        self.label_cant.setGeometry(QtCore.QRect(30, 89, 91, 19))
        font = QtGui.QFont()
        font.setPointSize(12)
        font.setBold(True)
        font.setWeight(75)
        self.label_cant.setFont(font)
        self.label_cant.setObjectName("label_cant")
        self.label_6.raise_()
        self.input_cant.raise_()
        self.btn_create.raise_()
        self.label_descrip.raise_()
        self.input_cod.raise_()
        self.label.raise_()
        self.label_categoria.raise_()
        self.btn_exit.raise_()
        self.label_precio.raise_()
        self.input_categoria.raise_()
        self.btn_cancel.raise_()
        self.input_descrip.raise_()
        self.label_cod.raise_()
        self.input_precio.raise_()
        self.label_cant.raise_()

        self.retranslateUi(CreateProduct)
        QtCore.QMetaObject.connectSlotsByName(CreateProduct)


          #--------------------Events--------------------------------------
        self.x = self.btn_create.clicked.connect(lambda:self.create_product_controller.createProduct(self.input_cod.text(),self.input_descrip.text(),self.input_cant.text(),self.input_precio.text(),self.input_categoria.text(),CreateProduct))
        
        self.c = self.btn_cancel.clicked.connect(lambda:self.create_product_controller.borrar())
        self.e = self.btn_exit.clicked.connect(lambda:self.create_product_controller.salir(CreateProduct))
        #--------------------End Events---------------------------------


    def retranslateUi(self, CreateProduct):
        _translate = QtCore.QCoreApplication.translate
        CreateProduct.setWindowTitle(_translate("CreateProduct", "Form"))
        self.input_cod.setPlaceholderText(_translate("CreateProduct", "código"))
        self.label_cod.setText(_translate("CreateProduct", "Código:"))
        self.input_cant.setPlaceholderText(_translate("CreateProduct", "cantidad"))
        self.btn_create.setText(_translate("CreateProduct", "Crear"))
        self.label_descrip.setText(_translate("CreateProduct", "Descripción:"))
        self.label.setText(_translate("CreateProduct", "Crear Nuevo Producto"))
        self.label_categoria.setText(_translate("CreateProduct", "Categoría:"))
        self.btn_exit.setText(_translate("CreateProduct", "Salir"))
        self.label_precio.setText(_translate("CreateProduct", "Precio:"))
        self.input_categoria.setPlaceholderText(_translate("CreateProduct", "categoría"))
        self.btn_cancel.setText(_translate("CreateProduct", "Cancelar"))
        self.input_descrip.setPlaceholderText(_translate("CreateProduct", "nombre"))
        self.input_precio.setPlaceholderText(_translate("CreateProduct", "precio"))
        self.label_cant.setText(_translate("CreateProduct", "Cantidad:"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    CreateProduct = QtWidgets.QWidget()
    ui = Ui_CreateProduct()
    ui.setupUi(CreateProduct)
    CreateProduct.show()
    sys.exit(app.exec_())
